onnxruntime/core/providers/cpu/tensor/upsample.cc

```diff
@@ -310,24 +310,46 @@ void UpsampleBilinear(int64_t batch_size,
                       AllocatorPtr& alloc,
                       GetOriginalCoordinateFunc get_original_coordinate) {
   std::vector<float> y_original;
+  y_original.reserve(output_height);
```

Comment:
best to reserve memory - we know the number of elements that we are going to put into this container.